### ISSUE TITLE
2.11.5: build fails in libfreerdp/core/info.c

### ISSUE THREAD

**Issue by kloczek** [FreeRDP/FreeRDP#9784]
<details>
<summary>Looks like something is wrong with new version</summary>

```console
[ 45%] Building C object libfreerdp/CMakeFiles/freerdp.dir/core/info.c.o
cd /home/tkloczko/rpmbuild/BUILD/FreeRDP-2.11.5/x86_64-redhat-linux-gnu/libfreerdp && /usr/bin/gcc -DBUILD_TESTING -DCAIRO_FOUND=1 -DCMAKE_BUILD_TYPE=RelWithDebInfo -DEXT_PATH=\"/usr/lib64/freerdp2/extensions\" -DFREERDP_EXPORTS -DFREERDP_TEST_EXPORTS -DHAVE_AT_LEAST_KRB_V1_13 -DHAVE_CONFIG_H -DWINPR_EXPORTS -DWITH_OPENSSL -DWITH_WAYLAND -DWITH_X11 -DWITH_XKBFILE -D_FILE_OFFSET_BITS=64 -Dfreerdp_EXPORTS -I/home/tkloczko/rpmbuild/BUILD/FreeRDP-2.11.5/x86_64-redhat-linux-gnu -I/home/tkloczko/rpmbuild/BUILD/FreeRDP-2.11.5/x86_64-redhat-linux-gnu/include -I/home/tkloczko/rpmbuild/BUILD/FreeRDP-2.11.5/include -I/home/tkloczko/rpmbuild/BUILD/FreeRDP-2.11.5/winpr/include -I/home/tkloczko/rpmbuild/BUILD/FreeRDP-2.11.5/x86_64-redhat-linux-gnu/winpr/include -I/usr/include/cairo -I/usr/include/ffmpeg -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -fPIC -Wall -Wno-unused-result -Wno-unused-but-set-variable -Wno-deprecated-declarations -Wimplicit-function-declaration -Wredundant-decls -g -fno-omit-frame-pointer -DWINPR_DLL -DNDEBUG -Wno-builtin-macro-redefined -D__FILE__='"x86_64-redhat-linux-gnu/libfreerdp/CMakeFiles/freerdp.dir/compiler_depend.ts"' -fPIC   -DWITH_GSSAPI -DWITH_GSSAPI_MIT -MD -MT libfreerdp/CMakeFiles/freerdp.dir/core/info.c.o -MF CMakeFiles/freerdp.dir/core/info.c.o.d -o CMakeFiles/freerdp.dir/core/info.c.o -c /home/tkloczko/rpmbuild/BUILD/FreeRDP-2.11.5/libfreerdp/core/info.c
/home/tkloczko/rpmbuild/BUILD/FreeRDP-2.11.5/libfreerdp/core/info.c: In function ‘rdp_read_info_null_string’:
/home/tkloczko/rpmbuild/BUILD/FreeRDP-2.11.5/libfreerdp/core/info.c:88:39: error: initialization of ‘const WCHAR *’ {aka ‘const short unsigned int *’} from incompatible pointer type ‘BYTE *’ {aka ‘unsigned char *’} [-Wincompatible-pointer-types]
   88 |                 const WCHAR* domain = Stream_Pointer(s);
      |                                       ^~~~~~~~~~~~~~
make[2]: *** [libfreerdp/CMakeFiles/freerdp.dir/build.make:863: libfreerdp/CMakeFiles/freerdp.dir/core/info.c.o] Error 1
```
</details>

**Comment by akallabeth** [comment]
@kloczek 
1. last change in that file was in version 2.9.0
2. this is just a warning, I doubt you could ever compile FreeRDP with `-Werror`

**Comment by kloczek** [comment]
> 2. this is just a warning, I doubt you could ever compile FreeRDP with `-Werror`

Please one more time check gcc arguments which I've provided.
I'm not using `-Werror`. I'm usimg only selective `-Werror=format-security` which is not affecting that result.
Only cause it could be could be use gcc 14.x from Fedora rawhide.

**Comment by kloczek** [comment]
Just tested and I can confirm that with new gcc 14.x 2.11.4 fails the same way.

**Comment by akallabeth** [comment]
some discussion about this at https://matrix.to/#/!mnXPZoPpWIGqYeWkOf:matrix.org/$v6PUkv7dphA3aXz042GMjDcclKzkMYMVkCQZHowQ_N8?via=matrix.org&via=tchncs.de&via=mozilla.org

tldr;

you need to add `-Wno-incompatible-pointer-types` if you´re using `gcc-14` or above

**Comment by kloczek** [comment]
Is it any issue with fixing code to clan this warning? 🤔

**Comment by akallabeth** [comment]
@kloczek yes.
